storage/remote/queue_manager.go

```diff
@@ -126,34 +136,35 @@ func NewStorageQueueManager(tsdb StorageClient, queueCapacity int) *StorageQueue
 				constLabels,
 			),
 			prometheus.GaugeValue,
-			float64(queueCapacity),
+			float64(cfg.QueueCapacity),
 		),
 	}
+
+	t.wg.Add(cfg.Shards)
```

Comment:
Originally had that, but caused a race in the test where one did:

```go
go q.Run()
q.Stop()
```

as the `Wait` in `Stop` could run before the `Add` in `Run`.